backend/app/Http/Requests/StoreCheckoutDetailRequest.php

```diff
@@ -24,8 +25,22 @@ public function authorize()
     public function rules()
     {
         return [
-            'checkout_id' => ['required','integer','min:1'],
-            'product_id' => ['required','integer','min:1'],
+            'checkout_id' => [
+                'required',
+                'integer',
+                'min:1', 
+                Rule::unique('checkout_details')
+                    ->where('order_id', $this->order_id)
+                    ->whereNull('deleted_at')
+            ],
+            'order_id' => [
+                'required',
+                'integer',
+                'min:1', 
+                Rule::unique('checkout_details')
+                    ->where('checkout_id', $this->checkout_id)
+                    ->whereNull('deleted_at')
+            ],
             'price' => ['required','integer','min:1']
```

Comment:
I think it's better when you make the price numerical instead of integer.
Please refer to this SO: https://stackoverflow.com/questions/40434642/laravel-validation-difference-between-numeric-and-integer

Reply:
Will do sir.